### PR TITLE
1a requests swb

### PR DESCRIPTION
**Pull request by Scott-Borecki**
### What I Did:
- Add User show endpoint:  #2 
- Add `json-apiserializer` gem
- Update User model, tests, and factory with name attributes
- Create RequestSpecHelper module (in `spec/support/`)
- Create UserSerializer
- Create shared examples for request specs (in `spec/support/shared_examples/`)
- Create ExceptionHandler and Response modules (in `/controllers/concerns/`)

### Road Blocks
- Adding self-referential relationships to UserSerializer.  This is my first time using the `json-apiserializer` gem, so didn't get to dig into the docs yet.

### Test Coverage
- [X] I have tested this code with RSpec

### Questions/Notes:
- We may need to add the `follower` and `followee` relationships to the serializer if those need to be passed through the endpoint.
- Still need to decide as a group whether we include `google_token` in the Users table.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -28,9 +28,10 @@ gem 'bootsnap', '>= 1.1.0', require: false
 # Use Rack CORS for handling Cross-Origin Resource Sharing (CORS), making cross-origin AJAX possible
 # gem 'rack-cors'
 
+gem 'date'
```
Comment (carolinectan) [inline Gemfile:31]: Nice job putting the gems in alphabetical order

File: spec/serializers/user_serializer_spec.rb [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+          expect(user_attributes.size).to eq(13)
+
+          expect(user_attributes).to have_key(:email)
+          expect(user_attributes[:email]).to be_a String
+
+          expect(user_attributes).to have_key(:google_id)
+          expect(user_attributes[:google_id]).to be_a String
+
+          expect(user_attributes).to have_key(:google_token)
+          expect(user_attributes[:google_token]).to be_a String
```
Comment (Scott-Borecki) [inline spec/serializers/user_serializer_spec.rb:39]: We need to decide whether we want to include the `google_token` in the Users table.

File: spec/serializers/user_serializer_spec.rb [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+          expect(user_attributes[:first_name]).to be_a String
+
+          expect(user_attributes).to have_key(:last_name)
+          expect(user_attributes[:last_name]).to be_a String
+
+          expect(user_attributes).to have_key(:full_name)
+          expect(user_attributes[:full_name]).to be_a String
+
+          # TODO: Add relationships for followers and followees?  Or will we do
+          #       that in a separate serialized hash?
```
Comment (Scott-Borecki) [inline spec/serializers/user_serializer_spec.rb:72]: We need to discuss as a group how we need to send `follower` and `followee` data through the endpoint and how that gets handled by the front end.

File: app/models/user.rb [hunk 4]
```diff
@@ -20,4 +16,7 @@ class User < ApplicationRecord
   validates :availability_morning, inclusion: { in: [true, false] }
   validates :availability_afternoon, inclusion: { in: [true, false] }
   validates :availability_evening, inclusion: { in: [true, false] }
+  validates :first_name, presence: true
+  validates :last_name, presence: true
+  validates :full_name, presence: true
```
[the remark this answers is not in the document]
Comment (carolinectan) [reply app/models/user.rb:21]: Good call. If the use of the name(s) is just for the dashboard, maybe we just use the email address?

File: spec/support/shared_examples/request_status_codes.rb [hunk 5]
```diff
@@ -0,0 +1,35 @@
+shared_examples 'status code 200' do
```
Comment (carolinectan) [inline spec/support/shared_examples/request_status_codes.rb:1]: nice way to DRY up tests

**Comment by carolinectan** [comment]
No merge conflicts. Good use of support/shared_examples to DRY up tests!

**Pull request merged**